vimiv/gui/thumbnail.py

```diff
@@ -143,12 +144,86 @@ def n_rows(self) -> int:
         """Return the number of rows."""
         return math.ceil(self.count() / self.n_columns())
 
+    def _first_rendered_index(self) -> int:
+        """Return the index of the first thumbnail to be rendered."""
+        if api.settings.thumbnail.load_behind.value == -1:
+            return 0
+        return max(0, self.current_index() - api.settings.thumbnail.load_behind.value)
+
+    def _last_rendered_index(self) -> int:
+        """Return the index of the last thumbnail to be rendered."""
+        if len(self._paths) == 0:
+            return 0
+        if api.settings.thumbnail.load_ahead.value == -1:
+            return len(self._paths) - 1
+        return min(
+            len(self._paths) - 1,
+            self.current_index() + api.settings.thumbnail.load_ahead.value,
+        )
+
+    def _rendered_index_range(self) -> Tuple[int, int]:
+        """Return the first and last indexes in a tuple fitting a Python range."""
+        return (self._first_rendered_index(), self._last_rendered_index() + 1)
+
+    def _rendered_thumbnail_pairs(self) -> List[Tuple[int, str]]:
+        """Return index/path pairs with those closest to current index first."""
+        current_index = self.current_index()
+        first_index, last_index = self._rendered_index_range()
+        pairs = list(
+            zip(range(first_index, last_index), self._paths[first_index:last_index])
+        )
+        pairs.sort(key=lambda p: abs(p[0] - current_index))
```

Comment:
Another random comment, should have probably started another review, sorry, there may be more coming :laughing: 

The combination of passing this as a list of tuples, which is later always unpacked and zipped seems .. over-the-top. I would definitely pass this as a dictionary instead, so we can later use `list(dict)`, `dict.items()`, and `dict.values()` instead of dealing with the list by hand.

First idea would be something along:
```python
indices = sorted(range(first_index, last_index), key=lambda i: abs(i - current_index))
return dict(zip(indices, self._paths[indices]))
```

Reply:
Your approach is definitely better. More concise, and probably more performant as well, without the extra unpacking. I just needed to replace `self._paths[indices]` with a list comprehension to select the appropriate paths.

Reply:
You are right of course, this is not a numpy array :smile: 

Although, we could then go straight for a dict-comprehension, but this is becoming super nit-picky:
```
        return {i: self._paths[i] for i in indices}
```

Reply:
I honestly didn't even know dict comprehensions were a thing. So, thank you for teaching me something new. Have another commit ready to push with the change, now.